src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java

```diff
@@ -0,0 +1,261 @@
+package net.snowflake.ingest.streaming.internal;
+
+import static java.math.RoundingMode.UNNECESSARY;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import net.snowflake.client.jdbc.internal.snowflake.common.util.Power10;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import net.snowflake.ingest.utils.Utils;
+import org.apache.parquet.schema.PrimitiveType;
+
+/** Parses a user column value into Parquet internal representation for buffering. */
+class ParquetValueParser {
+  /** Parquet internal value representation for buffering. */
+  static class ParquetBufferValue {
+    private final Object value;
+    private final float size;
+
+    ParquetBufferValue(Object value, float size) {
+      this.value = value;
+      this.size = size;
+    }
+
+    Object getValue() {
+      return value;
+    }
+
+    float getSize() {
+      return size;
+    }
+  }
+
+  /**
+   * Parses a user column value into Parquet internal representation for buffering.
+   *
+   * @param value column value provided by user in a row
+   * @param columnMetadata column metadata
+   * @param typeName Parquet primitive type name
+   * @param stats column stats to update
+   * @return parsed value and byte size of Parquet internal representation
+   */
+  static ParquetBufferValue parseColumnValueToParquet(
+      Object value,
+      ColumnMetadata columnMetadata,
+      PrimitiveType.PrimitiveTypeName typeName,
+      RowBufferStats stats) {
+    Utils.assertNotNull("Arrow column stats", stats);
```

Comment:
"Arrow" -> "Parquet"

Reply:
We also have a similiar problem in `ingest_eror_messages.properties` #L9 with `0004=The given row cannot be converted to Arrow format: {0}. {1}`